core/include/moveit/task_constructor/container_p.h

```diff
@@ -151,7 +158,9 @@ class ContainerBasePrivate : public StagePrivate
 	container_type children_;
 
 	// map start/end states of children (internal) to corresponding states in our external interfaces
-	std::map<const InterfaceState*, InterfaceState*> internal_to_external_;
+	boost::bimap<boost::bimaps::unordered_set_of<const InterfaceState*>,
```

Comment:
Why using a set of internal states here? Each internal state still has a unique external counterpart.
However, we have multiple internal stages mapping to the same external one in case of parallel containers, where an external stages is copied to all children.

Reply:
Sorry, I should have had a deeper look into the bimap doc before. So far, I only knew about `boost::bimap<A, B>`, but didn't have seen the collection-type specifications.